[bug] cmake_find_package_multi generator should not enforce that certain buildtypes are used.

### Environment Details (include every applicable attribute)
  * Conan version: 1.26.1

### Steps to reproduce (Include if Applicable)

I'm using cmake-conan but i'd assume this is preproduceable with other means too. 

CMakeLists.txt has:
```cmake
  set(CMAKE_CONFIGURATION_TYPES "Debug;RelWithDebInfo")
  conan_cmake_run(
    CONANFILE ${ROOT_CONANFILE}
    BASIC_SETUP
    UPDATE
    CMAKE_TARGETS
    CONFIGURATION_TYPES "${CMAKE_CONFIGURATION_TYPES}"
    GENERATORS "cmake_find_package_multi"
```

### Logs (Executed commands with output) (Include/Attach if Applicable)

```
-- Conan: Loading conanbuildinfo_multi.cmake
CMake Error at build/conanbuildinfo_multi.cmake:3488 (message):
  No conanbuildinfo_release.cmake, please install the Release conf first
Call Stack (most recent call first):
  cmake/conan.cmake:464 (include)
  cmake/conan.cmake:507 (conan_load_buildinfo)
  CMakeLists.txt:27 (conan_cmake_run)
```

### Afterthoughts 

For me, it it feels like that FATAL_ERROR messages are not really necessary, conanbuildinfo_$buildtype.cmake files are generated based upon what user provides and if release or debug is not on that list, they shouldnt be enforced to be present. 

In my usecase, this would actually save a minute from initial cmake run on 1gb network and cache on fast m2 ssd.

--- a/conans/client/generators/cmake_common.py
+++ b/conans/client/generators/cmake_common.py
@@ -966,25 +966,12 @@ cmake_macros = "\n".join([
 
 cmake_macros_multi = "\n".join([
     textwrap.dedent("""
-        if(EXISTS ${CMAKE_CURRENT_LIST_DIR}/conanbuildinfo_release.cmake)
-            include(${CMAKE_CURRENT_LIST_DIR}/conanbuildinfo_release.cmake)
-        else()
-            message(FATAL_ERROR "No conanbuildinfo_release.cmake, please install the Release conf first")
-        endif()
-
-        if(EXISTS ${CMAKE_CURRENT_LIST_DIR}/conanbuildinfo_debug.cmake)
-            include(${CMAKE_CURRENT_LIST_DIR}/conanbuildinfo_debug.cmake)
-        else()
-            message(FATAL_ERROR "No conanbuildinfo_debug.cmake, please install the Debug conf first")
-        endif()
-
-        if(EXISTS ${CMAKE_CURRENT_LIST_DIR}/conanbuildinfo_minsizerel.cmake)
-            include(${CMAKE_CURRENT_LIST_DIR}/conanbuildinfo_minsizerel.cmake)
-        endif()
-
-        if(EXISTS ${CMAKE_CURRENT_LIST_DIR}/conanbuildinfo_relwithdebinfo.cmake)
-            include(${CMAKE_CURRENT_LIST_DIR}/conanbuildinfo_relwithdebinfo.cmake)
-        endif()
+        ### load generated conanbuildinfo files.
+        foreach(_name release debug minsizerel relwithdebinfo)
+            if(EXISTS ${CMAKE_CURRENT_LIST_DIR}/conanbuildinfo_${_name}.cmake)
+                include(${CMAKE_CURRENT_LIST_DIR}/conanbuildinfo_${_name}.cmake)
+            endif()
+        endforeach()
         """),
     CMakeCommonMacros.conan_set_vs_runtime_preserve_build_type,
     CMakeCommonMacros.conan_set_find_paths_multi,

